Update eks cluster configuration (#1048)
rancher/rancher-docs: docs/reference-guides/cluster-configuration/rancher-server-configuration/eks-cluster-configuration.md

| Labels | Kubernetes labels applied to the nodes in the managed node group. |
| Tags | These are tags for the managed node group and do not propagate to any of the associated resources. |

### Self-managed Amazon Linux Nodes

Rancher can import an EKS cluster with self-managed Amazon Linux nodes. This type of cluster needs to be configured based on AWS documentation for launching self-managed Amazon Linux nodes. EKS clusters with self-managed Amazon Linux nodes are usually operated by the Karpenter project. After provisioning an EKS cluster with self-managed Amazon Linux nodes, it can be easily imported into Rancher Manager and then managed by it. However, the nodes won't be visible in Rancher Manager.

> ### IAM roles for service accounts

Reviewer: ```suggestion
### IAM Roles for Service Accounts
```

Author: I fixed that

> Applications Deployments running on EKS cluster can make requests to AWS services via IAM permissions. These applications must sign their requests with AWS credentials. IAM roles for service accounts manage these credentials using AWS OIDC endpoint. Rather than distributing AWS credentials to containers or relying on an EC2 instance's role, you can link an [IAM role to a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) and configure your Pods to use this account.

Reviewer: ```suggestion
An Applications Deployment running on an EKS cluster can make requests to AWS services via IAM permissions. These applications must sign their requests with AWS credentials. IAM roles for service accounts manage these credentials using an AWS OIDC endpoint. Rather than distributing AWS credentials to containers or relying on an EC2 instance's role, you can link an [IAM role to a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) and configure your Pods to use this account.
```

Author: Fixed

> An Applications Deployment running on an EKS cluster can make requests to AWS services via IAM permissions. These applications must sign their requests with AWS credentials. IAM roles for service accounts manage these credentials using an AWS OIDC endpoint. Rather than distributing AWS credentials to containers or relying on an EC2 instance's role, you can link an [IAM role to a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) and configure your Pods to use this account.

Reviewer 2: I know it says this is for **application deployments** but it might be worth adding a note somewhere to say this isn't applicable/supported for the Rancher pods when Rancher Manager is running in an EKS scluster.

Author: Added note about that. Thank you

> Enable IAM roles for service accounts by completing the following procedures:

Reviewer: ```suggestion
To enable IAM roles for service accounts:
```

Author: Fixed

| Labels | Kubernetes labels applied to the nodes in the managed node group. |
| Tags | These are tags for the managed node group and do not propagate to any of the associated resources. |

### Self-managed Amazon Linux Nodes

> Rancher can import an EKS cluster with self-managed Amazon Linux nodes. This type of cluster needs to be configured based on AWS documentation for launching self-managed Amazon Linux nodes. EKS clusters with self-managed Amazon Linux nodes are usually operated by the Karpenter project. After provisioning an EKS cluster with self-managed Amazon Linux nodes, it can be easily imported into Rancher Manager and then managed by it. However, the nodes won't be visible in Rancher Manager.

Reviewer: ```suggestion
Rancher can import an EKS cluster with self-managed Amazon Linux nodes. This type of cluster needs to be configured based on [AWS documentation for launching self-managed Amazon Linux nodes](https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html). EKS clusters with self-managed Amazon Linux nodes are usually operated by the [Karpenter](https://karpenter.sh/docs/) project. After you provision an EKS cluster with self-managed Amazon Linux nodes, the cluster can be easily imported into Rancher Manager and then managed by it. However, the nodes won't be visible in Rancher Manager.
```

> Rancher can import an EKS cluster with self-managed Amazon Linux nodes. This type of cluster needs to be configured based on [AWS documentation for launching self-managed Amazon Linux nodes](https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html). EKS clusters with self-managed Amazon Linux nodes are usually operated by the [Karpenter](https://karpenter.sh/docs/) project. After you provision an EKS cluster with self-managed Amazon Linux nodes, the cluster can be easily imported into Rancher Manager and then managed by it. However, the nodes won't be visible in Rancher Manager.

Reviewer: ```suggestion
You can register an EKS cluster containing self-managed Amazon Linux nodes. You must configure this type of cluster according to the instructions in the official AWS documentation for [launching self-managed Amazon Linux nodes](https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html). EKS clusters containing self-managed Amazon Linux nodes are usually operated by the [Karpenter](https://karpenter.sh/docs/) project. After you provision an EKS cluster containing self-managed Amazon Linux nodes, [register the cluster](../how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/register-existing-clusters.md) so it can be managed by Rancher. However, the nodes won't be visible in the Rancher UI.
```

Register vs import -- our [docs](https://ranchermanager.docs.rancher.com/v2.8/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/register-existing-clusters) say that "register" is the term that's replaced "import" but we're not very consistent about maintaining that distinction.

> You can register an EKS cluster containing self-managed Amazon Linux nodes. You must configure this type of cluster according to the instructions in the official AWS documentation for [launching self-managed Amazon Linux nodes](https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html). EKS clusters containing self-managed Amazon Linux nodes are usually operated by the [Karpenter](https://karpenter.sh/docs/) project. After you provision an EKS cluster containing self-managed Amazon Linux nodes, [register the cluster](../how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/register-existing-clusters.md) so it can be managed by Rancher. However, the nodes won't be visible in the Rancher UI.

Reviewer: ```suggestion
You can register an EKS cluster containing self-managed Amazon Linux nodes. You must configure this type of cluster according to the instructions in the official AWS documentation for [launching self-managed Amazon Linux nodes](https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html). EKS clusters containing self-managed Amazon Linux nodes are usually operated by the [Karpenter](https://karpenter.sh/docs/) project. After you provision an EKS cluster containing self-managed Amazon Linux nodes, [register the cluster](../../../how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/register-existing-clusters.md) so it can be managed by Rancher. However, the nodes won't be visible in the Rancher UI.
```

### IAM Roles for Service Accounts

An Applications Deployment running on an EKS cluster can make requests to AWS services via IAM permissions. These applications must sign their requests with AWS credentials. IAM roles for service accounts manage these credentials using an AWS OIDC endpoint. Rather than distributing AWS credentials to containers or relying on an EC2 instance's role, you can link an [IAM role to a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) and configure your Pods to use this account.

:::note

> It isn't supported for the Rancher pods when Rancher Manager is running in an EKS cluster.

Reviewer: On my first pass I had difficulty parsing this. After a few more reads, I think it means, Linking to an IAM role is not supported for Rancher pods when Rancher is running in EKS?

I think we need to replace the "It" in the note with a clear statement of what isn't supported.

:::

To enable IAM roles for service accounts:

> 1. [Create an IAM OIDC provider for your cluster](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
> 2. [Configure a Kubernetes service account to assume an IAM role](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html)
> 3. [Configure Pods to use a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html)
> 4. [Use a supported AWS SDK](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html)

Reviewer: ```suggestion
1. [Create an IAM OIDC provider for your cluster](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
1. [Configure a Kubernetes service account to assume an IAM role](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html)
1. [Configure Pods to use a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html)
1. [Use a supported AWS SDK](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html)
```

Just some nitpicky syntax (Markdown will automatically turn these into the proper numbers in case we need to add, remove, or move steps)
